t/21-needles.t

```diff
@@ -168,4 +180,69 @@ subtest 'handling relative paths in update_needle' => sub {
     };
 };
 
+subtest 'controller->_determine_needles_dir_for_job' => sub {
+    my $controller = OpenQA::WebAPI::Controller::Step->new;
+    $controller->app($t->app);
+    $settings{CASEDIR} = 'https://something#fragment';
+    my $sandbox = tempdir(CLEANUP => 1);
+    my $job = $schema->resultset('Jobs')->create_from_settings(\%settings);
+    $job->update({result_dir => $sandbox});
+
+    subtest 'checkout_needles_sha = no' => sub {
+        $controller->app->config->{'scm git'} = {checkout_needles_sha => 'no'};
+        my ($needle_dirs, $needles_var) = $controller->_determine_needles_dir_for_job($job);
+        is($needle_dirs, undef, 'needles directory determined');
+        is($needles_var, undef, 'needles var determined');
+    };
+
+    subtest 'checkout_needles_sha = yes' => sub {
+        $controller->app->config->{'scm git'} = {checkout_needles_sha => 'yes'};
+        my ($needle_dirs, $needles_var) = $controller->_determine_needles_dir_for_job($job);
+        like($needle_dirs, qr|t/data/openqa/share/tests/fedora/needles$|, 'needles dirs is defined');
+        is($needles_var->value, 'https://something#fragment', 'needles var is defined');
+    };
+};
+
+subtest 'controller->_create_tmpdir_for_needles_refspec' => sub {
+    my $tempdir = setup_fullstack_temp_dir('21-needles');
+    my $testresult_dir = $tempdir . '21-needles/openqa/testresults/00000';
+    path($testresult_dir)->make_path;
```

Comment:
There is already a method that does that for you:
```suggestion
    my $testresult_dir = $job->create_result_dir;
```
If you compare your variable and this one, you will see the important difference.

Reply:
Yep, thanks!